### ISSUE TITLE
Dataclasses and nested dataclasses/models

### ISSUE THREAD

**Issue by pauleveritt** [samuelcolvin/pydantic#273]
# Question

For bugs/questions:
* OS: macOS
* python 3.7
* pydantic 0.14

The following gives an error saying NavbarButton doesn't have a validator. I am marking this as a 
question because I think the docs explain the new dataclass support doesn't support validators of validators.

```py
from pydantic.dataclasses import dataclass


@dataclass
class NavbarButton:
    href: str


@dataclass
class Navbar:
    button: NavbarButton


if __name__ == '__main__':
    button = NavbarButton(href='a')
    navbar = Navbar(button=button)
```

**Comment by samuelcolvin** [comment]
I guess pydantic should be able to code with dataclasses, shouldn't be too hard to implement.

**Comment by Gr1N** [comment]
@samuelcolvin I would like to add support for nested dataclasses, but have some questions about implementation:
1) Should we support vanilla `dataclasses.dataclass` or only wrapped using `pyndatic.dataclasses.dataclass`?
2) If we're going to support vanilla dataclasses, how we must validate them? Just try to coerce or wrap into `pydantic`'s dataclass?

**Comment by pauleveritt** [comment]
@Gr1N I think it's fine to make it a restriction that, starting from a pydantic dataclass, the constructed nested dataclasses are pydantic ones.

**Comment by samuelcolvin** [comment]
fixed in #334, if you want to allow vanillia dataclasses, that already works with `arbitrary_types_allowed = True` otherwise pydantic dataclasses will be required.

**Comment by pauleveritt** [comment]
Super, thanks!
